Mathlib/Tactic/Simproc/Apply.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Lean.Elab.Tactic.Basic
+import Lean.Meta.Tactic.Simp.Main
+
+/-!
+# Procedure to execute a custom Simproc
+
+The only function `Simproc.apply` defined in this file takes a `s : Simproc` and executes it on the
+goal.
+-/
+
+open Lean Elab Tactic
+
+/-- Execute the given `Simproc` on the goal. -/
```

Comment:
This version executes it on the goal, but it's reasonable for someone to want it at a hypothesis, or at multiple hypotheses, or a hypothesis and the goal, etc. So perhaps we should mimic what `simp` does here, allowing it to run at multiple places.

Reply:
done